phpBB/phpbb/notification/type/forum.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\notification\type;
+
+/**
+* Forum notifications class
+* This class handles notifications for replies to a topic in a forum user subscribed to
+*/
+
+class forum extends \phpbb\notification\type\post
+{
+	/**
+	* Get notification type name
+	*
+	* @return string
+	*/
+	public function get_type()
+	{
+		return 'notification.type.forum';
+	}
+
+	/**
+	* Notification option data (for outputting to the user)
+	*
+	* @var bool|array False if the service should use it's default data
+	* 					Array of data (including keys 'id', 'lang', and 'group')
+	*/
+	static public $notification_option = [
+		'lang'	=> 'NOTIFICATION_TYPE_FORUM',
+		'group'	=> 'NOTIFICATION_GROUP_POSTING',
+	];
+
+	/**
+	* Find the users who want to receive notifications
+	*
+	* @param array $post Data from submit_post
+	* @param array $options Options for finding users for notification
+	*
+	* @return array
+	*/
+	public function find_users_for_notification($post, $options = [])
+	{
+		$options = array_merge([
+			'ignore_users'		=> [],
+		], $options);
+
+		$users = [];
+
+		$sql = 'SELECT user_id
+			FROM ' . FORUMS_WATCH_TABLE . '
+			WHERE forum_id = ' . (int) $post['forum_id'] . '
+				AND notify_status = ' . NOTIFY_YES . '
+				AND user_id <> ' . (int) $post['poster_id'];
+		$result = $this->db->sql_query($sql);
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			$users[] = (int) $row['user_id'];
+		}
+		$this->db->sql_freeresult($result);
+
+		$notify_users = $this->get_authorised_recipients($users, $post['forum_id'], $options, true);
+
+		if (empty($notify_users))
+		{
+			return [];
+		}
+
+		// Try to find the users who already have been notified about replies and have not read the topic since and just update their notifications
```

Comment:
Topic? :D

Reply:
Actually yes, as notification takes the user to unread posts in the topic.